code/object/object.cpp

```diff
@@ -1530,6 +1530,12 @@ void obj_move_all(float frametime)
 				// physics
 				obj_move_call_physics(objp, frametime);
 			}
+		} else if (objp->flags[Object::Object_Flags::Immobile]) {
+			// make sure velocity is always 0 for immobile things!
+			vm_vec_zero(&objp->phys_info.vel);
+			vm_vec_zero(&objp->phys_info.desired_vel);
+			vm_vec_zero(&objp->phys_info.rotvel);
+			vm_vec_zero(&objp->phys_info.desired_rotvel);
```

Comment:
This will zero out the velocity even when hull strength is <= 0.  I suggest rearranging the conditions so that the `Immobile` flag is only checked once.

Reply:
I definitely misread the condition, but this still doesn't apply to dying immobile objects, because they will pass the first condition and go into the first block.